app/connector/kafka/src/main/java/io/syndesis/connector/kafka/KafkaMetaDataRetrieval.java

```diff
@@ -15,45 +15,124 @@
+     * Used to filter which types of connections are we interested in. Right now, only plain connections.
+     */
+    private final Predicate<? super Map<?, ?>>
+        typesAllowed = listener -> listener.get("type").toString().equalsIgnoreCase("plain");
+
     /**
      * TODO: use local extension, remove when switching to camel 2.22.x
      */
     @Override
-    protected MetaDataExtension resolveMetaDataExtension(CamelContext context, Class<? extends MetaDataExtension> metaDataExtensionClass, String componentId, String actionId) {
+    protected MetaDataExtension resolveMetaDataExtension(CamelContext context,
+                                                         Class<? extends MetaDataExtension> metaDataExtensionClass,
+                                                         String componentId, String actionId) {
         return new KafkaMetaDataExtension(context);
     }
 
     @SuppressWarnings("unchecked")
     @Override
-    protected SyndesisMetadata adapt(CamelContext context, String componentId, String actionId, Map<String, Object> properties, MetaDataExtension.MetaData metadata) {
+    protected SyndesisMetadata adapt(CamelContext context, String componentId, String actionId,
+                                     Map<String, Object> properties, MetaDataExtension.MetaData metadata) {
         try {
             Set<String> topicsNames = (Set<String>) metadata.getPayload();
 
             List<PropertyPair> topicsResult = new ArrayList<>();
-            topicsNames.stream().forEach(
+            topicsNames.forEach(
                 t -> topicsResult.add(new PropertyPair(t, t))
             );
 
             return SyndesisMetadata.of(
                 Collections.singletonMap("topic", topicsResult)
             );
-        } catch ( Exception e) {
+        } catch (Exception e) {
             return SyndesisMetadata.EMPTY;
         }
     }
 
+    /**
+     * Query the strimzi brokers available on this kubernetes environment to suggest auto discovered urls.
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public SyndesisMetadataProperties fetchProperties(CamelContext context, String componentId,
+                                                      Map<String, Object> properties) {
+        List<PropertyPair> brokers = new ArrayList<>();
+        try (KubernetesClient client = new DefaultKubernetesClient()) {
+
+            //Filter by strimzi resources
+            final CustomResourceDefinitionContext build = new CustomResourceDefinitionContext.Builder()
+                                                              .withGroup("kafka.strimzi.io")
+                                                              .withName("kafkas.kafka.strimzi.io")
+                                                              .withPlural("kafkas")
+                                                              .withScope("Namespaced")
+                                                              .withVersion("v1beta1")
```

Comment:
Can we make `version` configurable or make the user supply the version instead of hardcoding it?

Reply:
The query requires a version to be supplied. You mean choosing from the UI? That may be not very understandable for the final user.

I am going to check if I can query a list of versions too...

Reply:
maybe we do not need the version anymore when refactoring to model deserialization and fetching the custom resource definition with the client.

Reply:
Aaah I left the other suggestion for later because it looked harder.

Reply:
Version is part of the request URL when accessing the CRDs resource via k8s API. We could list and detect what CRDs are present and choose version from that.